js/files.ts

```diff
@@ -29,14 +29,51 @@ export const stdout = new File(1);
 /** An instance of `File` for stderr. */
 export const stderr = new File(2);
 
-// TODO This is just a placeholder - not final API.
-export type OpenMode = "r" | "w" | "w+" | "x";
+export enum OpenMode {
+  /** Read-only, default mode.
+   * Starts at beginning of file.
+   */
+  Read = "r",
+  /** Read-write
+   * Start at beginning of file.
+   */
+  ReadWrite = "r+",
+  /** Write-only
+   * Opens and truncates existing file or creates new one
+   * for writing only.
+   */
+  Write = "w",
+  /** Write-read
+   * Opens and truncates existing file or creates new one
+   * for writing and reading.
+   */
+  WriteRead = "w+",
```

Comment:
The name of this one is strange. `WriteTruncate` would be better ?

Reply:
Get rid of names completely.